LetsDoStuff.WebApi/Controllers/ParticipationController.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using LetsDoStuff.WebApi.Services;
+using LetsDoStuff.WebApi.Services.DTO;
+using LetsDoStuff.WebApi.Services.Interfaces;
+using LetsDoStuff.WebApi.SettingsForAuthJwt;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LetsDoStuff.WebApi.Controllers
+{
+    [Route("api/participation")]
+    public class ParticipationController : ControllerBase
+    {
+        private readonly IParticipationService _participationService;
+
+        public ParticipationController(IParticipationService participationService)
+        {
+            _participationService = participationService;
+        }
+
+        /// <summary>
+        /// Take participation in an activity.
+        /// </summary>
+        /// <param name="idActivity">The Id of an Activity.</param>
+        /// <returns>Action result.</returns>
+        [HttpGet("AddParticipation")]
+        [Authorize]
+        public IActionResult AddParticipation(int idActivity)
+        {
+            try
+            {
+                var idUser = int.Parse(this.HttpContext.User.Claims
+                    .Where(c => c.Type == UserClaimIdentity.DefaultIdClaimType)
+                    .First().Value);
+                _participationService.AddParticipation(idUser, idActivity);
+                return Ok();
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Refuse to participation in an Activity.
+        /// </summary>
+        /// <param name="idActivity">The Id of an Activity.</param>
+        /// <returns>Action result.</returns>
+        [HttpGet("RemoveParticipation")]
```

Comment:
the same, but httpDelete

Reply:
[HttpDelete]
[Authorize]
public IActionResult RemoveParticipation(ParticipationRequest request) ...